### PR TITLE
Update set-up-database.rst

### PR DESCRIPTION
**Pull request by hanifmusa96**
Update procedure to setup PostgreSQL database

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: docs/apache-airflow/howto/set-up-database.rst [hunk 1]
```diff
@@ -167,6 +167,7 @@ In the example below, a database ``airflow_db`` and user  with username ``airflo
    CREATE USER airflow_user WITH PASSWORD 'airflow_pass';
    GRANT ALL PRIVILEGES ON DATABASE airflow_db TO airflow_user;
    -- PostgreSQL 15 requires additional privileges:
+   \c airflow_db;
```
Comment (kaxil) [inline docs/apache-airflow/howto/set-up-database.rst]: This isn't SQL statement

**Review by kaxil — CHANGES_REQUESTED** [review]

File: docs/apache-airflow/howto/set-up-database.rst (hunk 1 above)
Comment (hanifmusa96) [reply docs/apache-airflow/howto/set-up-database.rst]: meta-command is acceptable in psql 
reference: https://www.postgresql.org/docs/current/app-psql.html

Comment (hanifmusa96) [reply docs/apache-airflow/howto/set-up-database.rst]: another extra reference:
- https://stackoverflow.com/questions/55590601/grant-privileges-to-a-specific-table-with-postgresql-relation-does-not-exis

Comment (kaxil) [reply docs/apache-airflow/howto/set-up-database.rst]: psql is a cli tool to interact with Postgres where you can run and do more things than query. The code block here is pure SQL statement.

Comment (kaxil) [reply docs/apache-airflow/howto/set-up-database.rst]: Meta-commands aren't SQL :)

Comment (kaxil) [reply docs/apache-airflow/howto/set-up-database.rst]: If we merge this, users who copy paste their code to their sql tools / automation -- will have failures

**Review by potiuk — APPROVED** [review]

**Review by kaxil — APPROVED** [review]

**Pull request merged**